esmvaltool/diag_scripts/aerosols/aero_utils.py

```diff
@@ -0,0 +1,270 @@
+
+    Parameters
+    ----------
+    wavel : str
+        A given wavelength.
+
+    Returns
+    -------
+    pseudo-level index : str
+        The pseudo-level index corresponding to the given wavelength.
+    """
+
+    # Wavelengths (nm)
+    cwavel = ['380', '440', '500', '675', '870', '1020']
+
+    try:
+        return cwavel.index(str(wavel)) + 1  # for 1: indexing in files
+
+    except AeroAnsError:
+        print('Get_Wavelen_Index: requested Wavelength not found' + str(wavel),
+              ' not in ', cwavel)
+
+
+def extract_pt(icube, pt_lat, pt_lon, **keys):
+    """Extracts given location(s) (3-D) from a cube.
+
+    Method
+    ------
+    Uses Iris module Analysis.Interpolate to extract values
+    first based on horizontal coordinates and later based on
+    height, if specified.
+
+    If height ('altitude') is requested, checks if cube heights
+    include orography, i.e. HybridHeights have been derived.
+
+    Parameters
+    ----------
+    icube : Iris cube
+    pt_lat, pt_lon : latitude and longitude coordinates of desired points
+    keys: (optional)
+        height  : altitude (above geoid) of point
+        level   : model/ pseudo/ tile number, default = all
+        nearest : (optional) Whether to use 'nearest neighbour', instead
+                  of 'linear' method while extracting data.
+
+    Returns
+    -------
+    data_out : List
+        List of single point values, corresponding to each point specified.
+    """
+
+    # Check that input data is a (single) cube
+
+    if not isinstance(icube, iris.cube.Cube):
+        raise AeroAnsError('Extract_pt:First argument must be a single cube')
+
+    # Check that equal number of lat/lon pairs are passed in point coordinates
+    # convert arguments to lists as easier to process later
+    pt_lat1 = []
+    pt_lon1 = []
+
+    if not isinstance(pt_lat, list):
+        pt_lat1.append(pt_lat)
+        pt_lon1.append(pt_lon)
+
+    else:
+        for n in np.arange(len(pt_lat)):
+            pt_lat1.append(pt_lat[n])
+            pt_lon1.append(pt_lon[n])
+
+    if len(pt_lat1) != len(pt_lon1):
+        raise AeroAnsError('Extract_pt:Mismatch in number of lat/long values')
+
+    if 'level' in keys and 'height' in keys:
+        raise AeroAnsError('Extract_pt: Both Level and Height requested')
+
+    if icube.coords()[0].name() == 'time':  # only lat,lon,lev accepted
+        raise AeroAnsError(
+            'Extract_pt:Cannot handle time dimension at present')
+
+    if 'level' in keys and not icube.coord(
+            'model_level_number') and not icube.coord('pseudo_level'):
+        raise AeroAnsError('Extract_pt:Level requested, but not found in cube')
+
+    if 'height' in keys:
+        pt_hgt = []
+
+        if not isinstance(keys['height'], list):
+            pt_hgt.append(keys['height'])
+
+        else:
+            for n in np.arange(len(keys['height'])):
+                pt_hgt.append(keys['height'][n])
```

Comment:
this whole block can be replaced with `pt_hgt.extend(keys['height'])`

Reply:
Done